Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -779,6 +834,47 @@ void DpCatalog ::pingIn_handler(FwIndexType portNum, U32 key) {
     this->pingOut_out(0, key);
 }
 
+void DpCatalog ::addToCat_handler(FwIndexType portNum,
+                                  const Fw::StringBase& fileName,
+                                  FwDpPriorityType priority,
+                                  FwSizeType size) {
+    // check some asserts
+    FW_ASSERT(this->m_dpTree);
+    FW_ASSERT(this->m_traverseStack);
+
+    // check initialization
+    if (not this->checkInit()) {
+        this->log_WARNING_HI_NotInitialized();
+        return;
+    }
+
+    // check that initialization got memory
+    if (0 == this->m_numDpSlots) {
+        this->log_WARNING_HI_NoDpMemory();
+        return;
+    }
+
+    // make sure a downlink is not in progress
+    if (this->m_xmitInProgress) {
```

Comment:
I think we should allow it to be added if the xmit is in progress. The downloads can potentially cover many hourse with a slow link.